### DISCUSSION TITLE
HTTPX_LOG_LEVEL=trace doesn't actually hook into the trace extension

### DISCUSSION THREAD

**Discussion by jashandeep-sohi** [encode/httpx#2036] [Potential Issue]
Per the docs, https://www.python-httpx.org/environment_variables/, I should be able to set set `HTTPX_LOG_LEVEL=trace` and it should log detailed information about HTTP requests.

However this does not work (at least for the latest version `0.21.3`). After a little digging, it looks like the only place the `trace` extension is actually hooked into is the CLI client: https://github.com/encode/httpx/blob/83c81aa9b8ef0f7825df413a673e6d9300065f3a/httpx/_main.py#L479

Is this a bug or are the docs wrong?

**Comment by lovelydinosaur** [comment]
Thanks for mentioning this. I've raised an issue at #2042
